fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanReadPlatformService.java

```diff
@@ -155,4 +150,17 @@ org.springframework.data.domain.Page<LoanTransactionData> retrieveLoanTransactio
     List<Long> retrieveLoanIdsByExternalIds(List<ExternalId> externalIds);
 
     boolean existsByLoanId(Long loanId);
+
+    Collection<LoanTermVariationsData> retrieveLoanTermVariations(Long loanId, Integer termType);
+
```

Comment:
Seems unrelated changes, please raise a separate PR!

Also drop `Collection` return `List` or `Set` based on whether you need ordering or not!